From 33260a40302da01eec659aa3fa6f9dc0ded8a9d5 Mon Sep 17 00:00:00 2001
From: Ioana Radulescu <ruxandra.radulescu@freescale.com>
Date: Tue, 19 Jun 2012 23:16:55 +0000
Subject: [PATCH 40/57] t4240: Add FMan support in device tree

commit fe48d94ecbac29b6d9cd3d9447b38bffa9d85d28 from
https://git.freescale.com/git-private/cgit.cgi/ppc/t4240/linux.git

Add minimal FMan support only (first 1G port of first FMan).
No MDIO support so using fixed-link instead.

Signed-off-by: Ioana Radulescu <ruxandra.radulescu@freescale.com>
---
 arch/powerpc/boot/dts/fsl/t4240si-post.dtsi |   10 ++++++++++
 arch/powerpc/boot/dts/fsl/t4240si-pre.dtsi  |    1 +
 arch/powerpc/boot/dts/t4240qds.dts          |   16 ++++++++++++++++
 3 files changed, 27 insertions(+)

diff --git a/arch/powerpc/boot/dts/fsl/t4240si-post.dtsi b/arch/powerpc/boot/dts/fsl/t4240si-post.dtsi
index 5c75206..e1f3668 100644
--- a/arch/powerpc/boot/dts/fsl/t4240si-post.dtsi
+++ b/arch/powerpc/boot/dts/fsl/t4240si-post.dtsi
@@ -269,4 +269,14 @@
 /include/ "qoriq-bman1.dtsi"
 
 /include/ "qoriq-rman-0.dtsi"
+
+/include/ "qoriq-fman3-0.dtsi"
+/include/ "qoriq-fman3-0-1g-0.dtsi"
+	fman0: fman@400000 {
+		/* tx - 1g - 0 */
+		port@a8000 {
+			fsl,qman-channel-id = <0x802>;
+		};
+	};
+
 };
diff --git a/arch/powerpc/boot/dts/fsl/t4240si-pre.dtsi b/arch/powerpc/boot/dts/fsl/t4240si-pre.dtsi
index d7307a5..06243b7 100644
--- a/arch/powerpc/boot/dts/fsl/t4240si-pre.dtsi
+++ b/arch/powerpc/boot/dts/fsl/t4240si-pre.dtsi
@@ -48,6 +48,7 @@
 		serial3 = &serial3;
 		bman = &bman;
 		qman = &qman;
+		fman0 = &fman0;
 		pci0 = &pci0;
 		pci1 = &pci1;
 		pci2 = &pci2;
diff --git a/arch/powerpc/boot/dts/t4240qds.dts b/arch/powerpc/boot/dts/t4240qds.dts
index 8449d82..4f99953 100644
--- a/arch/powerpc/boot/dts/t4240qds.dts
+++ b/arch/powerpc/boot/dts/t4240qds.dts
@@ -202,6 +202,14 @@
 				interrupts = <0x1 0x1 0 0>;
 			};
 		};
+
+		 fman0: fman@400000 {
+			enet0: ethernet@e0000 {
+				fixed-link = <1 1 1000 0 0 >;
+				phy-connection-type = "sgmii";
+				local-mac-address = [02 01 01 01 01 01];
+			};
+		 };
 	};
 
 	pci0: pcie@ffe240000 {
@@ -273,6 +281,14 @@
 			ranges = <0 0 0xc 0x30000000 0 0x10000000>;
 		};
 	};
+
+	fsl,dpaa {
+		compatible = "fsl,t4240-dpaa", "fsl,dpaa";
+		ethernet@0 {
+			compatible = "fsl,t4240-dpa-ethernet", "fsl,dpa-ethernet";
+			fsl,fman-mac = <&enet0>;
+		};
+	};
 };
 
 /include/ "fsl/t4240si-post.dtsi"
-- 
1.7.9.7

